src/crewai/conditional_task.py

```diff
@@ -0,0 +1,35 @@
+from typing import Callable, Optional, Any
+from crewai.task import Task
+from crewai.tasks.task_output import TaskOutput
+
+
+class ConditionalTask(Task):
+    """
+    A task that can be conditionally executed based on the output of another task.
+    Note: This cannot be the only task you have in your crew and cannot be the first since its needs context from the previous task.
+    """
+
+    condition: Optional[Callable[[TaskOutput], bool]] = None
```

Comment:
Why is this optional? Should a conditional task have to have a condition or else it would just be a regular task.